include/YTArchive.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import unicode_literals; import youtube_dl
+
+class MyLogger(object): # prints out warning and error messages
+    def debug(self, msg): pass
+    def warning(self, msg): print(msg)
+    def error(self, msg): print(msg)
+def my_hook(d):
+    if d['status'] == 'finished': print('Done downloading... start converting')  
+def downer(code, link): # standard video and playlist specs for Hydrus
+    ydl_opts = {
+        'format': code,
+        'logger': MyLogger(), # see class above
+        'progress_hooks': [my_hook], # see function above
+        'write_all_thumbnails':True, # forced thumbnail for easy searching
+        'writesubtitles': 'allsubtitles', # forced subtitle for easy searching
+        'outtmpl': 'yt-dl/%(title)s/%(title)s-%(format_id)s.%(ext)s'
+    }
+    if not isinstance(link, list): link = [link] # conversion to list
+    with youtube_dl.YoutubeDL(ydl_opts) as ydl: ydl.download(link)
+
+def set_pick(dash, container="mp4", resolution=[360,720], hfr=False):
+    assert dash in [True, False, None], "error: dash is not bool or None"
+    assert container in ["mp4", "webm", "both"], \
+        "error: container variable not correct"
+    assert set(resolution) < set([240, 360, 480, 720, 1080]), \
+        "error: bad input from resolution"
+    assert isinstance(hfr, bool), "error: hfr is not boolean"
+    non_dash = [[18,22],[43]]; dash_audio = [[140], [171, 249, 250, 251]]
+    dash_video = [
+        #240,  360,  480,  720, 1080, 720h,1080h
+        [133,  134,  135,  136,  137,  288,  289],
+        [242,  243,  244,  247,  248,  302,  303]]
+    dash_video_flip = list(zip(*dash_video)) # reversing x and y axis
+    table = {240:[0], 360:[1], 480:[2,5], 720:[3], 1080:[4,6]}
+    unity = sum(dash_audio, []) + sum(dash_video, []) # all dash-media
+    royale = unity + sum(non_dash, []) # all media
+    def format_war(n): return dash_audio[n] + dash_video[n] + non_dash[n]
```

Comment:
This isn't a `.war` file, so what does the `war` refer to?

You may have wanted `zip` here instead of defining your own function.

Reply:
the war was used as a synonym for "standards" or "specs" in this case. Also good on you for using zip.